experimental/alp/include/alp/Transforms/Passes.td

```diff
@@ -52,4 +54,14 @@ def LegalizePass: FunctionPass<"alp-legalize"> {
   ];
 }
 
+def ForToDoWhileLoop: Pass<"alp-for-to-dowhile", "FuncOp"> {
+  let summary = "Pass to legalize vector operations.";
```

Comment:
It is interesting to see such a transformation is needed.
Could you please document and explain why it is necessary?

Reply:
Extracted as a standalone #257 but I was not able to get extra perf from it.
Would be nice to have a use case that exhibits the benefit.
Is there a specific configuration you could share ?

Reply:
In some cases it does not make much difference. We found it particularly useful when working with dynamic shapes: the difference is about 30%, so quite an impact. It might also boil down to how llvm applies passes. Maybe for x86 it always converts, while it does not for AArch64?

Reply:
Dynamic is prob the keyword here.
We can experiment more later.